spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -32,21 +33,37 @@
  * @since 1.0
  *
  */
-public class UnZipResultSplitter {
-
-	public List<Message<Object>> splitUnzippedMap(Map<String, Object> unzippedEntries) {
+public final class UnZipResultSplitter {
 
-		final List<Message<Object>> messages = new ArrayList<Message<Object>>(unzippedEntries.size());
+	public List<Message<Object>> splitUnzippedMap(Message<Map<String, Object>> message) {
+		return createMessages(message.getPayload(), message.getHeaders());
+	}
 
+	private List<Message<Object>> createMessages(Map<String, Object> unzippedEntries, MessageHeaders headers) {
+		List<Message<Object>> messages = createList(unzippedEntries.size());
 		for (Map.Entry<String, Object> entry : unzippedEntries.entrySet()) {
-			final String path = FilenameUtils.getPath(entry.getKey());
-			final String filename = FilenameUtils.getName(entry.getKey());
-			final Message<Object> splitMessage = MessageBuilder.withPayload(entry.getValue())
-					.setHeader(FileHeaders.FILENAME, filename)
-					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path).build();
-			messages.add(splitMessage);
+			messages.add(createMessage(entry, headers));
 		}
 		return messages;
 	}
 
+	private static List<Message<Object>> createList(int initialCapacity) {
+		return new ArrayList<Message<Object>>(initialCapacity);
+	}
```

Comment:
@artembilan yes, the purpose of this method is just creation of `List<...>` instance. Moreover, it's possible to make `createList` as `protected` and class user can define it's own `List<...>` implementation without changing the main method in his subclass.

So for me introducing this method is a step forward to make whole class more flexible. But it's another question how this class should be implemented to be reusable, flexible and extendable. And definitely it's not a purpose of my PR. So if you think `createList` isn't needed here - remove it.